lib/model/document.rb

```diff
@@ -227,6 +227,28 @@ def handle_populations
           population['title'] = "Population #{index+1}"
         end
       end
+
+      # Remove exclusion and exception populations that are empty. The simple
+      # contains these empty populations, while the HQMF does not.
+      @populations.each do |population|
+        populations_to_remove = []
+        population.each do |key, value|
+          if ['DENEX', 'DENEXCEP', 'NUMEX', 'MSRPOPLEX'].include?(key)
+            criteria = @population_criteria.find { |pop_crit| pop_crit.id == value }
```

Comment:
I find the role of `population_criteria` vs `population` tricky to remember, especially how `population` here is the `{'NUMEX' -> 'numex2'}` mapping (or however it's actually done).
Consider adding some comments to clarify what these objects both are (maybe this isn't the right spot for it, but I think it would be good to include somewhere in the file a description of `populations` and `population_criteria`).